pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -232,6 +277,20 @@ func (b *volumeBinder) BindPodVolumes(assumedPod *v1.Pod) error {
 		if err != nil {
 			// only revert assumed cached updates for volumes we haven't successfully bound
 			b.revertAssumedPVs(bindings[i:])
+			// Revert all of the assumed cached updates for claims,
+			// since no actual API update will be done
+			b.revertAssumedPVCs(claimsToProvision)
+			return err
+		}
+	}
+
+	// Update claims objects to trigger volume provisioning. Let the PV controller take care of the rest
+	// PV controller is expect to signal back by removing related annotations if actual provisioning fails
+	for i, claim := range claimsToProvision {
+		if _, err := b.ctrl.kubeClient.CoreV1().PersistentVolumeClaims(claim.Namespace).Update(claim); err != nil {
```

Comment:
Since you update PVC, should you add it to PVC cache? Is it possible that `revertAssumedPVCs()` called by later iteration reverts to a version that's already old and API server has a new one?

Reply:
>Since you update PVC, should you add it to PVC cache?

I'm quite sure about your meaning here, we should have already add it to the assume cache in the previous assume phase

>Is it possible that revertAssumedPVCs() called by later iteration reverts to a version that's already old and API server has a new one?

I guess it will not happen, in the Restore func of the assume cache, we won't try to add an API obj into the cache, but change its `latestObj` same to its `apiObj`

Reply:
RevertAssumedPVCs called later only reverts PVCs that haven't been API updated yet `i:`

Reply:
The "provision-api-update-failed" unit test checks that 1st PVC shows update in cache, but 2nd PVC that failed API update got reverted in cache